### PR TITLE
support multiple distros in upgrade test

### PR DESCRIPTION
**Pull request by amoskong**

### CODE REVIEW

**Comment by roydahan** [comment]
let's merge this one as well.

File: upgrade_test.py [hunk 1]
```diff
@@ -70,34 +70,56 @@ def upgrade_node(self, node):
...
+                result = node.remoter.run("sudo yum search scylla-enterprise", ignore_status=True)
+                new_is_enterprise = True if ('scylla-enterprise.x86_64' in result.stdout or
+                                             'No matches found' not in result.stdout) else False
+            else:
+                result = node.remoter.run("sudo apt-cache search scylla-enterprise", ignore_status=True)
+                new_is_enterprise = True if 'scylla-enterprise' in result.stdout else False
+
             scylla_pkg = 'scylla-enterprise' if new_is_enterprise else 'scylla'
-            if orig_is_enterprise != new_is_enterprise:
+            if orig_is_enterprise != new_is_enterprise: # or not node.is_rhel_like():
```
Comment (bentsi) [inline upgrade_test.py]: is this comment necessary?

File: upgrade_test.py [hunk 2]
```diff
@@ -110,34 +132,51 @@ def rollback_node(self, node):
...
+            node.remoter.run('sudo apt-get update')
 
         if re.findall('\d+.\d+', self.orig_ver)[0] == re.findall('\d+.\d+', self.new_ver)[0]:
             self.upgrade_rollback_mode = 'minor_release'
 
-        if self.upgrade_rollback_mode == 'reinstall':
-            node.remoter.run('sudo yum remove scylla\* -y')
-            node.remoter.run('sudo yum install %s -y' % node.scylla_pkg())
-            node.remoter.run('for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ); do sudo cp -v $conf.autobackup $conf; done')
+        if self.upgrade_rollback_mode == 'reinstall' or not node.is_rhel_like():
```
Comment (bentsi) [inline upgrade_test.py]: are you sure that for all `not node.is_rhel_like()` you will reinstall Scylla?

Comment (bentsi) [reply upgrade_test.py]: In other words when OS is not rhel_like you will reinstall Scylla and never get to the code where
`self.upgrade_rollback_mode == 'minor_release'`

File: upgrade_test.py [hunk 3]
```diff
@@ -110,34 +132,51 @@ def rollback_node(self, node):
...
+            if node.is_rhel_like():
+                node.remoter.run('sudo yum remove scylla\* -y')
+                node.remoter.run('sudo yum install %s -y' % node.scylla_pkg())
+                node.remoter.run('for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ); do sudo cp -v $conf.autobackup $conf; done')
+            else:
+                node.remoter.run('sudo apt-get remove scylla\* -y')
+                node.remoter.run('sudo apt-get install %s -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" --force-yes --allow-unauthenticated' % node.scylla_pkg())
+                node.remoter.run('for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles /var/lib/dpkg/info/scylla-*conf.conffiles /var/lib/dpkg/info/scylla-*jmx.conffiles | grep -v init ); do sudo cp -v $conf.backup $conf; done')
+                node.remoter.run('sudo systemctl daemon-reload')
         elif self.upgrade_rollback_mode == 'minor_release':
```
Comment (bentsi) [inline upgrade_test.py]: this flow is checked only for rhel like OS-s. Is it intended so?

File: upgrade_test.py (hunk 2 above)
Comment (amoskong) [reply upgrade_test.py]: For ubuntu/debian, In rollback we always remove the packages first, then reinstall original scylla packages.

check the doc: https://docs.scylladb.com/upgrade/upgrade-guide-from-2.2-to-2.3-ubuntu/

File: upgrade_test.py (hunk 3 above)
Comment (amoskong) [reply upgrade_test.py]: We always use reinstall mode for ubuntu/debian rollback.
So the minor_release and smooth mode are only for RHEL.

**Pull request merged**